Sources/StreamCore/Logger/Logger.swift

```diff
@@ -237,66 +308,74 @@ public enum LogConfig {
         }
     }
     
-    private nonisolated(unsafe) static var _destinations: [LogDestination]?
+    private static let _destinations = AllocatedUnfairLock<[LogDestination]?>(nil)
```

Comment:
I think we can remove this one. Any time the logger is being recreated (which isn't that often) we can create the destinations again. As a benefit we remove a locking point from the process.